### ISSUE TITLE
Automatically zoom in on report pin

### ISSUE THREAD

**Issue by jonkri** [mysociety/fixmystreet#1649]
A FixaMinGata user has made a suggestion that could be a UX improvement for the FixMyStreet platform.

When a user clicks on a report, the map pin becomes larger. The request is to also have the map zoom in on the report. The benefit of this would be to more clearly see where on the map the report is.

If this behaviour was realized, then the map could also be made to automatically zoom back out when the navigating back to the report list.

I'm not sure what should happen if the user has panned or zoomed the map while viewing the report, but my spontanous idea is that the map should be reset to the way it was prior to clicking on the report initially, since any zooming or panning seems unlikely to be relevant at the point at which the user hits the back button anyway.

**Comment by altinukshini** [comment]
I thought the same until I figured out that the second click on the problem pin would zoom you in on the report. But as you mentioned, after that, if clicking "Back to all reports" doesn't zoom out the map.

**Comment by jonkri** [comment]
Interesting,  @altinukshini! Thanks!

I guess that the fact that the back button won't work should be considered defect, then?

As a sidenote: When I said zooming, I imagined the zoom to be animated, but I have no idea what that would mean from a technical perspective!

**Comment by altinukshini** [comment]
I've figured out that if you open a fresh page for a report, let's say **/report/40** and hover over the "Back to all reports" link, you can see a **&zoom=X** parameter added at the end of the url. 
That is the value you specify in your cobrand module under **default_link_zoom**
And only in this case this feature works as supposed!

It doesn't work when you are viewing few reports under /around* > click on a pin (to zoom in, or just show the problem report) and then try to go back to all reports by clicking "Back to all reports". The zoom level parameter is still there in the "Back to all reports" link, but it doesn't get triggered - not sure why.

**Comment by jonkri** [comment]
@dracos: Is the above behaviour to be considered a bug, you think?

**Comment by jonkri** [comment]
A user requested the same zoom level when clicking on a map pin (or in the problem list) as when following the e-mail link to a problem report. Also, it would probably be best if the back button restored the zoom level.

**Comment by jonkri** [comment]
I would like to suggest that FixMyStreet is changed to restore the original zoom level when the user navigates back to all reports.

A common activity for staff users is to zoom in on reports, and then wanting to go back to see all reports. Zooming out becomes time-consuming, and an automatic reset of the zoom would allow the staff users to save time. Also, the text "Back to all reports" indicates that all available reports will be displayed.

Here are some steps that illustrate the current behaviour:

1. Visit https://beta.fixamingata.se/reports/Ale
2. Click on a report pin - the size of the pin changes and the report is displayed
3. Click again on the same pin - the zoom level changes
4. Click Back, or on Tillbaka till samtliga rapporter (Back to all reports) - the zoom level remains the same

The suggested behaviour would be that when Step 4 is carried out, the zoom level is adjusted to the way it was in Step 1.

Thank you!

**Comment by jonkri** [comment]
@dracos: Hi! I just noticed the Next label. Is this being worked on? :) Thanks!

**Comment by jonkri** [comment]
Trying out FixMyStreet.com today, I behaviour of the zoom level surprised me, when using the Back button in combination with report filters. In case this is unintended, a bug report follows.

Perhaps this should be a separate issue?

### Description of problem

The Back button does not reset the zoom level when used in combination with report filters. This causes its behaviour to differ from the way that the Back to all reports link works. This behaviour has been reproduced on both the mobile site (on iPhone) and on the desktop site (on Firefox).

### Version

The live version of FixMyStreet.

### Steps to reproduce

1. Visit All reports
2. Under Pick your council, pick City of London Corporation
3. Scroll down, toggle the "Open" dropdown menu, and deselect Open and select Fixed
4. Scroll up and tap twice on a map pin (which causes the report to be zoomed in)
5. Click the native Back button

### Actual results

The zoom level remains zoomed in, as in Step 4.

### Expected results

The zoom level changes to what it was prior to Step 4 (whether or not the map has been zoomed or panned after Step 2).

### How reproducible

Always.

**Comment by dracos** [comment]
Yeah, sounds like changing the filter isn't then remembering the initial map zoom state.

**Comment by jonkri** [comment]
@dracos: By the way: Would it, when this is done, be possible to update Transifex so that the latest strings can be translated? Thank you!

**Comment by dracos** [comment]
This should be fixed (again).
I'll be doing a 2.5 release this week at some point, which will include the updated files, hope that's okay?

**Comment by jonkri** [comment]
This week works great! Thanks a lot! 🙂
